### ISSUE TITLE
[Bug]:  Problem when OCR heavy PDFs - freezes at 0%

### ISSUE THREAD

**Issue by dariofilipe** [ocrmypdf/OCRmyPDF#1116]
### Describe the bug

When i try to OCR PDFs that have heavy images with text (ex: 10mb or 20mb each image), it freezes at 0% and never moves. I tested out with a PDF with 70MB size. Are there any limitations that i'm missing? (ex: related with size, etc). What can i do to avoid this error for big files?

NOTE: After a while of being freezed, it generates the pdf. but it takes too long. See log output below. Also, i wanted to put the PDF here for you to test it, but github only allows 25MB max size. You can access from here: https://we.tl/t-4wkv7rrZPf

### Steps to reproduce

```plain text
1. Run ocrmypdf ~/Downloads/image.pdf test.pdf --force-ocr --optimize 0 --output-type pdf --fast-web-view 0
```

### OCRmyPDF version

13.4.0+dfsg

### Relevant log output

```plain text
Scanning contents: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 6/6 [00:00<00:00, 297.72page/s]
Start processing 6 pages concurrently
    4 [tesseract] lots of diacritics - possibly poor OCR                                                                                                                                                           
    4 [tesseract] Image too small to scale!! (2x36 vs min width of 3)                                                                                                                                              
    4 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    4 [tesseract] Image too small to scale!! (2x36 vs min width of 3)                                                                                                                                              
    4 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    4 [tesseract] Image too small to scale!! (2x36 vs min width of 3)                                                                                                                                              
    4 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    4 [tesseract] Image too small to scale!! (2x36 vs min width of 3)                                                                                                                                              
    4 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    4 [tesseract] Image too small to scale!! (1x36 vs min width of 3)                                                                                                                                              
    4 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    1 [tesseract] lots of diacritics - possibly poor OCR                                                                                                                                                           
    1 [tesseract] Image too small to scale!! (2x36 vs min width of 3)                                                                                                                                              
    1 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    1 [tesseract] Image too small to scale!! (2x36 vs min width of 3)                                                                                                                                              
    1 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    1 [tesseract] Image too small to scale!! (2x36 vs min width of 3)                                                                                                                                              
    1 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    1 [tesseract] Image too small to scale!! (2x36 vs min width of 3)                                                                                                                                              
    1 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    1 [tesseract] Image too small to scale!! (1x36 vs min width of 3)                                                                                                                                              
    1 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    3 [tesseract] lots of diacritics - possibly poor OCR                                                                                                                                                           
    3 [tesseract] Image too small to scale!! (2x36 vs min width of 3)                                                                                                                                              
    3 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    3 [tesseract] Image too small to scale!! (2x36 vs min width of 3)                                                                                                                                              
    3 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    3 [tesseract] Image too small to scale!! (2x36 vs min width of 3)                                                                                                                                              
    3 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    3 [tesseract] Image too small to scale!! (2x36 vs min width of 3)                                                                                                                                              
    3 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    3 [tesseract] Image too small to scale!! (1x36 vs min width of 3)                                                                                                                                              
    3 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    2 [tesseract] lots of diacritics - possibly poor OCR                                                                                                                                                           
    2 [tesseract] Image too small to scale!! (2x36 vs min width of 3)                                                                                                                                              
    2 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    2 [tesseract] Image too small to scale!! (2x36 vs min width of 3)                                                                                                                                              
    2 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    2 [tesseract] Image too small to scale!! (2x36 vs min width of 3)                                                                                                                                              
    2 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    2 [tesseract] Image too small to scale!! (2x36 vs min width of 3)                                                                                                                                              
    2 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    2 [tesseract] Image too small to scale!! (1x36 vs min width of 3)                                                                                                                                              
    2 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    5 [tesseract] lots of diacritics - possibly poor OCR                                                                                                                                                           
    5 [tesseract] Image too small to scale!! (2x36 vs min width of 3)                                                                                                                                              
    5 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    5 [tesseract] Image too small to scale!! (2x36 vs min width of 3)                                                                                                                                              
    5 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    5 [tesseract] Image too small to scale!! (2x36 vs min width of 3)                                                                                                                                              
    5 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    5 [tesseract] Image too small to scale!! (2x36 vs min width of 3)                                                                                                                                              
    5 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    5 [tesseract] Image too small to scale!! (1x36 vs min width of 3)                                                                                                                                              
    5 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    6 [tesseract] lots of diacritics - possibly poor OCR                                                                                                                                                           
    6 [tesseract] Image too small to scale!! (2x36 vs min width of 3)                                                                                                                                              
    6 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    6 [tesseract] Image too small to scale!! (2x36 vs min width of 3)                                                                                                                                              
    6 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    6 [tesseract] Image too small to scale!! (2x36 vs min width of 3)                                                                                                                                              
    6 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    6 [tesseract] Image too small to scale!! (2x36 vs min width of 3)                                                                                                                                              
    6 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
    6 [tesseract] Image too small to scale!! (1x36 vs min width of 3)                                                                                                                                              
    6 [tesseract] Line cannot be recognized!!                                                                                                                                                                      
OCR: 100%|█████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 6.0/6.0 [03:02<00:00, 30.45s/page]
Postprocessing...
```

**Comment by jbarlow83** [comment]
I believe the issue was fixed in 13.4.1 - a deadlock can occur in older versions. With the current main branch, the file is processed correctly (but does take time, given its size). Took about 1 minute on a 16 core system.

**Comment by jbarlow83** [comment]
See the documentation for options on upgrading to a new version than Debian/Ubuntu provide.
